concepts/references/about.md

```diff
@@ -1,3 +1,168 @@
 # About
 
-TODO
+To understand references it is necessary to understand ownership. Every value has a single owner, which is in a scope. When the scope is exited,
```

Comment:
> Every value has a single owner, which is in a scope.

I'm still not happy with that. Even if other documentation puts ownership at the variable level, we have an opportunity here to teach people more precisely what the language is doing.

- Values can have a **binding**. This is what is created with the `let`, `const`, and `static` statements. It identifies the value.
- Function parameters are also bindings which may or may not own the underlying value.
- `const` and `static` bindings can appear at the module level because they are valid for the entire lifetime of the program.
- `let` bindings can only appear within a defined scope.

What counts as a scope?

- functions, including methods and closures
- curly brace pairs which appear within functions

What happens at the end of a scope?

- References get returned to whatever previously held them. 
  - This may or may not be the owner, because a function holding a reference can pass that reference along to an internal call.
  - Due to non-lexical lifetimes, references are sometimes returned before the end of the scope. See [docs](https://doc.rust-lang.org/edition-guide/rust-2018/ownership-and-lifetimes/non-lexical-lifetimes.html).
- Owned items are freed
  - If the owned item implements `Drop`, that function is called first

The major point here is that the _scope_ is what governs when an item is freed, not the binding. A binding may be shadowed, but the shadowed value still lives until the end of the scope. 

Therefore, I strongly desire that we teach that the scope is the fundamental owner, not the binding.

```suggestion
To understand references it is necessary to understand ownership. Every value has a single owner, which is a scope. When the scope is exited,
```

Reply:
Just saw this comment. I'm gonna have to literally sleep on it. Of course the authors of TRPL are not infallible, but my first instinct is to maintain consistency with what is taken as closest to **the** reference of the language. But I'm really too tired to think about it right now.

Reply:
Sorry you had to wait so long for a reply. I seem to be fighting something off that may be more than just fatigue, plus I got my first jab a couple days ago.

It's true that ownership can't exist without bindings _and_ scope. But it seems that the non-lexical lifetime example in the docs indicates that ownership more particularly follows a binding and how it is used than scope per se. As the last line of the "Non-lexical lifetimes" section says: "Instead of pointing to where y goes out of scope, it shows you where the conflicting borrow occurs." That's why we can have two mutable borrows in the same scope, as long as they don't conflict with each other. It has more to do with how two the bindings interact in time than what scope they're in. If I bring a ball onto a playground you could say the playground contains the ball, but I'm holding the ball. If I give the ball to someone else they now hold the ball. It happens in the playground, and the ball is in the playground, but it's the people who hold the ball.

Reply:
No worries; we're all doing this in what free time we can spare. It's certainly not unreasonable to take half a day to think!

I wrote an [example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=6ce7dfa04bd9a1b81a174ce22564c272) of what I'm talking about. NLL only applies to references within a scope. The ball in playground metaphor is nice, and mostly works most of the time, but it doesn't accurately describe what actually happens.

Reply:
I'm only skimming this discussion so my apologies for not weighing in.

I am curious about the example.

Is the block scope of `fn main` being torn down the reason why the shadowed variable `"second"` is dropped first, followed by the `"first"`? like Rust is unwinding or something?

Reply:
@bobahop feel better! 

To the core work here: how do @coriolinus and @bobahop you both feel about approving and merging the current work?

I can see both sides to the discussion. I would vote for one or more follow-up pull requests to clarify our teaching semantics.

Reply:
I mean, there's a reason I've been using the "comment" feature instead of the "request changes" feature: I don't mean to actively block progress if you (or another maintainer) feels that I'm being obstreperous here. I've intentionally not actively vetoed the current state of things. However, I will not approve it until I feel it is correct.

Reply:
I better understand where you're coming from with your `Drop` example. 

I was intrigued by this statement in the [Drop](https://doc.rust-lang.org/std/ops/trait.Drop.html#you-cannot-call-dropdrop-yourself) reference that `Drop::drop does not take ownership of its input`, and wondered how relevant to ownership `Drop` is vs. its relevance to lifetime and scope (although all three are intertwined.)

I'm not sure I'm sold that if the binding were the owner, we'd see dropping first and then dropping second. I can interpret that as a variable stacking its shadow, so when it goes out of scope, the shadow owner pops off first and then the original owner. But I freely admit I don't know how it's implemented under the covers.

Well, I don't want this to be filioque-type schism here.

I wonder if you'd be okay with `Every value has a single owner. When its scope is exited, the value is dropped.` It only mentions scope and not `variable`, as the TRPL does in `Each value in Rust has a variable that’s called its owner.`

Reply:
As I understand it, the reason `Drop` doesn't take ownership of its input is to ensure that it's never recursive. 

I'm fine with that proposed phrasing.